File: C/codechef/Easy/Ciel and Receipt.c
#include<stdio.h>

int main()
{
    short t, temp;
    int num, menus;
    scanf("%hd", &t);
    while(t--)
    {
        scanf("%d", &num);
        temp = 2048;
        menus = 0;

        while(num > 0)
        {
            while (num >= temp)
            {
                num-=temp;
                menus++;
            }
            temp /= 2;
            if(temp == 0)
                temp = 1;
        }
        printf("%d\n", menus);
    }
    return 0;
}

File: Python2/codechef/easy/Ciel and Receipt.py
import sys
def p():
    cases = int(sys.stdin.readline())
    for case in xrange(cases):
        num = int(sys.stdin.readline())
        temp = 2048
        menus = 0

        while num > 0:
            while (num - temp) >= 0:
                num -= temp
                menus += 1

            temp /= 2
            if temp == 0:
                temp = 1

        print menus

p()
        
